crates/burrego/src/opa/wasm.rs

```diff
@@ -0,0 +1,440 @@
+        LoadedPolicies {
+            policies: Vec::new(),
+        }
+    }
+
+    fn register(&mut self, policy_name: String) -> usize {
+        self.policies.push(policy_name);
+        self.policies.len() - 1
+    }
+
+    fn policy(&self, policy_id: usize) -> Option<String> {
+        self.policies.get(policy_id).cloned()
+    }
+}
+
+use lazy_static::lazy_static;
+lazy_static! {
+    static ref LOADED_POLICIES: RwLock<LoadedPolicies> = RwLock::new(LoadedPolicies::new());
+    static ref BUILTINS_HELPER: RwLock<BuiltinsHelper> = {
+        let lookup_tables: HashMap<usize, LookupTable> = HashMap::new();
+
+        RwLock::new(BuiltinsHelper {
+            builtins: builtins::get_builtins(),
+            lookup_tables,
+        })
+    };
+}
+
+pub struct Evaluator {
+    #[allow(dead_code)]
+    engine: Engine,
+    #[allow(dead_code)]
+    linker: Linker<Option<StackHelper>>,
+    store: Store<Option<StackHelper>>,
+    instance: Instance,
+    memory: Memory,
+    policy: Policy,
+}
+
+impl Evaluator {
+    pub fn from_path(policy_name: String, policy_path: &Path) -> Result<Evaluator> {
+        Evaluator::new(policy_name, &std::fs::read(&policy_path)?)
+    }
+
+    pub fn new(policy_name: String, policy_contents: &[u8]) -> Result<Evaluator> {
+        let engine = Engine::default();
+        let mut linker = Linker::<Option<StackHelper>>::new(&engine);
+
+        let opa_data_helper: Option<StackHelper> = None;
+        let mut store = Store::new(&engine, opa_data_helper);
+
+        let memory_ty = MemoryType::new(Limits::new(5, None));
+        let memory = Memory::new(&mut store, memory_ty)?;
+        linker.define("env", "memory", memory)?;
+
+        // env.opa_abort(addr) void
+        // Called if an internal error occurs.
+        // The addr refers to a null-terminated string in the shared memory buffer.
+        let opa_abort = Func::wrap(
+            &mut store,
+            move |mut caller: Caller<'_, Option<StackHelper>>, addr: i32| {
+                let stack_helper = caller.data().unwrap();
+                let msg = stack_helper
+                    .pull_json(caller.as_context_mut(), &memory, addr)
+                    .unwrap();
+                println!("OPA abort with message: {:?}", msg);
+                std::process::exit(1);
+            },
+        );
+        linker.define("env", "opa_abort", opa_abort)?;
+
+        //env.opa_println (addr) void
+        //Called to emit a message from the policy evaluation.
+        //The addr refers to a null-terminated string in the shared memory buffer.
+        let opa_println = Func::wrap(
+            &mut store,
+            move |mut caller: Caller<'_, Option<StackHelper>>, addr: i32| {
+                let stack_helper = caller.data().unwrap();
+                let msg = stack_helper
+                    .pull_json(caller.as_context_mut(), &memory, addr)
+                    .unwrap();
+                println!("Message coming from the policy: {:?}", msg);
```

Comment:
Same here, we will probably have to build a callback mechanism. The consumer will provide a function/closure to use to handle `abort` and `println`